diff --git a/source/frontends/common2/config.h.in b/source/frontends/common2/config.h.in
index ffe5c851..1d4a3cf9 100644
--- a/source/frontends/common2/config.h.in
+++ b/source/frontends/common2/config.h.in
@@ -3,4 +3,4 @@
 #cmakedefine SHARE_PATH "@SHARE_PATH@"
 
 // this one is a bit of a hack, until resources are embedded in the retro core
-#cmakedefine CMAKE_SOURCE_DIR "@CMAKE_SOURCE_DIR@"
+#cmakedefine CMAKE_SOURCE_DIR "/opt/applewin"
