diff --git a/source/frontends/sdl/utils.cpp b/source/frontends/sdl/utils.cpp
index edca8e81..ef124149 100644
--- a/source/frontends/sdl/utils.cpp
+++ b/source/frontends/sdl/utils.cpp
@@ -45,7 +45,7 @@ namespace sa2
 
         const int iscapture = 0;
 
-#if SDL_VERSION_ATLEAST(3, 0, 0)
+/*#if SDL_VERSION_ATLEAST(3, 0, 0)
         // not relevant in SDL3
 #elif SDL_VERSION_ATLEAST(2, 24, 0)
         char *defaultDevice = nullptr;
@@ -56,7 +56,7 @@ namespace sa2
             SDL_free(defaultDevice);
         }
         else
-#endif
+#endif*/
 
         {
             os << "<default>";
