diff --git a/Makefile b/Makefile
index 0ee72599..69402efb 100644
--- a/Makefile
+++ b/Makefile
@@ -258,22 +258,22 @@ else ifeq ($(platform), classic_armv7_a7)
 else ifeq ($(platform), classic_armv8_a35)
 	EXT    ?= so
 	TARGET := $(TARGET_NAME)_libretro.$(EXT)
-	SHARED += -shared -Wl,--version-script=link.T
+	SHARED += -shared -Wl,--version-script=link.T -Wl,--no-undefined
 	fpic = -fPIC
 	LIBS += -lrt
 	ARM_FLOAT_ABI_HARD = 1
 	FORCE_GLES = 1
 	SINGLE_PREC_FLAGS = 1
 	HAVE_LTCG = 0
-	HAVE_OPENMP = 0
+	HAVE_OPENMP = 1
 	CFLAGS += -Ofast \
-	-fuse-linker-plugin \
 	-fno-stack-protector -fno-ident -fomit-frame-pointer \
 	-fmerge-all-constants -ffast-math -funroll-all-loops \
 	-marm -mcpu=cortex-a35 -mfpu=neon-fp-armv8 -mfloat-abi=hard
 	CXXFLAGS += $(CFLAGS)
-	ASFLAGS += $(CFLAGS)
-	LDFLAGS += -marm -mcpu=cortex-a35 -mfpu=neon-fp-armv8 -mfloat-abi=hard -Ofast -flto -fuse-linker-plugin
+	ASFLAGS += $(CFLAGS) -c -DHOST_CPU=0x20000002 -mfpu=neon-fp-armv8 -mfloat-abi=hard -DARM_HARDFP
+	CC_AS = $(CC)
+	LDFLAGS += -marm -mcpu=cortex-a55 -mfpu=neon-fp-armv8 -mfloat-abi=hard -Ofast
 	PLATFORM_EXT := unix
 	WITH_DYNAREC = arm
 	HAVE_GENERIC_JIT = 0
