diff --git a/core/libretro/libretro_core_options.h b/core/libretro/libretro_core_options.h
index 3433ba40..8a6de33f 100644
--- a/core/libretro/libretro_core_options.h
+++ b/core/libretro/libretro_core_options.h
@@ -228,9 +228,9 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       {
          { "per-strip (fast, least accurate)", "Per-Strip (fast, least accurate)" },
          { "per-triangle (normal)",            "Per-Triangle (normal)" },
-#if defined(HAVE_OIT) || defined(HAVE_VULKAN)
+//#if defined(HAVE_OIT) || defined(HAVE_VULKAN)
          { "per-pixel (accurate)",             "Per-Pixel (accurate, but slowest)" },
-#endif
+//#endif
          { NULL, NULL },
       },
 #if defined(LOW_END)
