diff --git a/Makefile b/Makefile
index 65796cf..9a7d68a 100644
--- a/Makefile
+++ b/Makefile
@@ -250,6 +250,17 @@ else ifeq ($(platform), ctr)
 	HAVE_DYNAREC = 1
 	STATIC_LINKING = 1
 
+# Go-Advance
+else ifeq ($(platform), goadvance)
+	CPUFLAGS := -marm -march=armv8-a+crc+simd -mtune=cortex-a35 -mcpu=cortex-a35 -mfpu=neon-fp-armv8 -mfloat-abi=hard -flto
+	CFLAGS  := $(CPUFLAGS) -fpic -ffast-math -fno-rtti -fno-exceptions
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,--version-script=link.T -Wl,--no-undefined
+	CPU_ARCH := arm
+    MMAP_JIT_CACHE = 1
+	HAVE_DYNAREC = 1
+
 # Raspberry Pi 3
 else ifeq ($(platform), rpi3)
 	TARGET := $(TARGET_NAME)_libretro.so
