diff --git a/libraries/ZWidget/src/window/sdl2/sdl2displaywindow.cpp b/libraries/ZWidget/src/window/sdl2/sdl2displaywindow.cpp
index 17db8a863..d685f3366 100644
--- a/libraries/ZWidget/src/window/sdl2/sdl2displaywindow.cpp
+++ b/libraries/ZWidget/src/window/sdl2/sdl2displaywindow.cpp
@@ -1,6 +1,8 @@
 
 #include "sdl2displaywindow.h"
 #include <stdexcept>
+#include <cstring>
+#include <cmath>
 
 Uint32 SDL2DisplayWindow::PaintEventNumber = 0xffffffff;
 bool SDL2DisplayWindow::ExitRunLoop;
