diff --git a/src/io/input.cpp b/src/io/input.cpp
index dc8fddc..078d9fe 100644
--- a/src/io/input.cpp
+++ b/src/io/input.cpp
@@ -144,10 +144,10 @@ int g_joy_btn_defs[SWITCH_COUNT][2] = {
 // Three values defined: 1st=joystick index (0=1st joystick), 2nd=axis index (0=not defined), 3rd=direction (-1/1, 0=not defined)
 // Only first four switch defined (SWITCH_UP to SWITCH_RIGHT)
 int g_joy_axs_defs[SWITCH_START1][3] = {
-    {0, 2, -1}, // up
-    {0, 1, -1}, // left
-    {0, 2, 1},  // down
-    {0, 1, 1}   // right
+    {0, 1, -1}, // up
+    {0, 2, -1}, // left
+    {0, 1, 1},  // down
+    {0, 2, 1}   // right
 };
 
 // Mouse button to key mappings
