diff --git a/Makefile b/Makefile
index 7b7978d..80d3295 100644
--- a/Makefile
+++ b/Makefile
@@ -174,7 +174,7 @@ else ifeq ($(platform), libnx)
 else ifneq (,$(findstring odroid64,$(platform)))
    TARGET := $(TARGET_NAME)_libretro.so
    LDFLAGS += -shared -Wl,--version-script=$(LIBRETRO_DIR)/link.T -Wl,--no-undefined
-   BOARD := $(shell cat /proc/cpuinfo | grep -i odroid | awk '{print $$3}')
+   BOARD := C2
    GLES = 1
    GL_LIB := -lGLESv2
    WITH_DYNAREC := aarch64
