diff --git a/src/platform/libretro/libretro.c b/src/platform/libretro/libretro.c
index 663674dd..b29697f4 100644
--- a/src/platform/libretro/libretro.c
+++ b/src/platform/libretro/libretro.c
@@ -1624,9 +1624,11 @@ void retro_run(void) {
 
 	if (rumbleCallback) {
 		if (rumbleUp) {
+			system("echo 10 > /sys/class/pwm/pwmchip0/pwm0/duty_cycle");
 			rumbleCallback(0, RETRO_RUMBLE_STRONG, rumbleUp * 0xFFFF / (rumbleUp + rumbleDown));
 			rumbleCallback(0, RETRO_RUMBLE_WEAK, rumbleUp * 0xFFFF / (rumbleUp + rumbleDown));
 		} else {
+			system("echo 1000000 > /sys/class/pwm/pwmchip0/pwm0/duty_cycle");
 			rumbleCallback(0, RETRO_RUMBLE_STRONG, 0);
 			rumbleCallback(0, RETRO_RUMBLE_WEAK, 0);
 		}
