diff --git a/src/input.cpp b/src/input.cpp
index cac7782..95d861b 100644
--- a/src/input.cpp
+++ b/src/input.cpp
@@ -123,8 +123,14 @@ bool input_init(void)
 
 void rumble(float str, uint32_t len)
 {
-  if (settings->rumble)
+  if (settings->rumble){
+    FILE *file;
+    if ((file = fopen("/sys/class/pwm/pwmchip0/pwm0/duty_cycle", "r"))) {
+      system("echo 10 > /sys/class/pwm/pwmchip0/pwm0/duty_cycle && sleep 0.12 && echo 1000000 > /sys/class/pwm/pwmchip0/pwm0/duty_cycle &");
+      fclose(file);
+    }
     SDL_JoystickRumble(joy, 0xFFFF * str, 0xFFFF * str, len);
+  }
 }
 
 // set the SDL key that triggers an input
