diff --git a/src/game.h b/src/game.h
index d3408d1..8bf5aac 100644
--- a/src/game.h
+++ b/src/game.h
@@ -10,7 +10,7 @@
 #include "statusbar.h"
 #include "tsc.h"
 
-#define GAME_FPS 50
+#define GAME_FPS 60
 
 // for UpdateBlockedStates
 #define RIGHTMASK 0x01
diff --git a/src/statusbar.cpp b/src/statusbar.cpp
index 7965159..c95acba 100644
--- a/src/statusbar.cpp
+++ b/src/statusbar.cpp
@@ -389,9 +389,9 @@ void niku_draw(int value, bool force_white)
 
   Renderer::getInstance()->sprites.drawSprite(NIKU_X, NIKU_Y, SPR_NIKU_CLOCK, clkframe);
 
-  int mins = (value / 3000); // the game runs at 50 fps
-  int secs = (value / 50) % 60;
-  int tens = (value / 5) % 10;
+  int mins = (value / 3600); // the game runs at 60 fps
+  int secs = (value / 60) % 60;
+  int tens = (value / 6) % 10;
 
   DrawNumber(NIKU_X, NIKU_Y, mins);
   DrawTwoDigitNumber(NIKU_X + 36, NIKU_Y, secs);
