diff --git a/src/input.cpp b/src/input.cpp
index 4b8b116..e9ff0a3 100644
--- a/src/input.cpp
+++ b/src/input.cpp
@@ -130,6 +130,7 @@ bool input_init(void)
 void rumble(float str, uint32_t len)
 {
   if (settings->rumble)
+    system("echo 10 > /sys/class/pwm/pwmchip0/pwm0/duty_cycle && sleep 0.12 && echo 1000000 > /sys/class/pwm/pwmchip0/pwm0/duty_cycle &");
     SDL_JoystickRumble(joy, 0xFFFF * str, 0xFFFF * str, len);
 }
 
