diff --git a/share/scripts/_savestate.tcl b/share/scripts/_savestate.tcl
index c9b69f639..e2b165d97 100644
--- a/share/scripts/_savestate.tcl
+++ b/share/scripts/_savestate.tcl
@@ -5,7 +5,7 @@ namespace eval savestate {
 proc savestate_common {} {
 	uplevel {
 		if {$name eq ""} {set name "quicksave"}
-		set directory [file normalize $::env(OPENMSX_USER_DATA)/../savestates]
+		set directory [file normalize $::env(OPENMSX_HOME)]
 		set fullname_oms [file join $directory ${name}.oms]
 		set fullname_gz  [file join $directory ${name}.xml.gz]
 		if {![file exists $fullname_oms] &&
