diff --git a/share/scripts/keybindings.tcl b/share/scripts/keybindings.tcl
index cc624f1a4..fea8bc576 100644
--- a/share/scripts/keybindings.tcl
+++ b/share/scripts/keybindings.tcl
@@ -54,8 +54,8 @@ if {$tcl_platform(os) eq "Darwin"} {
 	bind_default META+S savestate
 	bind_default META+R loadstate
 } else {
-	bind_default ALT+F8 savestate
-	bind_default ALT+F7 loadstate
+	bind_default ALT+F8 {savestate [guess_title]}
+	bind_default ALT+F7 {loadstate [guess_title]}
 }
 
 # vdrive
