diff --git a/share/scripts/osd_menu.tcl b/share/scripts/osd_menu.tcl
index 24cdc849f..3a6fb74e4 100644
--- a/share/scripts/osd_menu.tcl
+++ b/share/scripts/osd_menu.tcl
@@ -48,7 +48,7 @@ proc update_button_fade {} {
 	after "mouse motion" [namespace code update_button_fade]
 }
 
-if {![regexp dingux "[openmsx_info platform]"]} {
+if {![regexp dingux "[openmsx_info platform]"] && 0} {
 	# add a button to pop up the menu
 	osd create rectangle main_menu_pop_up_button -z 0 -x 0 -y 0 -w 35 -h 16 -rgba $default_bg_color -fadePeriod $button_fadeout_time
 	osd create text main_menu_pop_up_button.text -z 0 -x 0 -y 0 -rgba $default_text_color -text "menu"
