diff --git a/frontend/libretro.c b/frontend/libretro.c
index 6fee42a..c9e8b87 100644
--- a/frontend/libretro.c
+++ b/frontend/libretro.c
@@ -495,8 +495,22 @@ void plat_trigger_vibrate(int pad, int low, int high)
 
    if (in_enable_vibration)
    {
-      rumble_cb(pad, RETRO_RUMBLE_STRONG, high << 8);
-      rumble_cb(pad, RETRO_RUMBLE_WEAK, low ? 0xffff : 0x0);
+      FILE *file;
+      if ((file = fopen("/sys/class/pwm/pwmchip0/pwm0/duty_cycle", "r+"))) {
+            if(high <= 0x80)
+            {
+              fputs("1000000", file);
+              fclose(file);
+            }
+            else
+            {
+              fputs("10", file);
+              fclose(file);
+            }
+      } else {
+         rumble_cb(pad, RETRO_RUMBLE_STRONG, high << 8);
+         rumble_cb(pad, RETRO_RUMBLE_WEAK, low ? 0xffff : 0x0);
+      }
    }
 }
 
