diff --git a/menu/drivers/ozone.c b/menu/drivers/ozone.c
index e0e071954e..d5c7b97770 100644
--- a/menu/drivers/ozone.c
+++ b/menu/drivers/ozone.c
@@ -7173,7 +7173,7 @@ static void *ozone_init(void **userdata, bool video_is_threaded)
    ozone->last_width        = width;
    ozone->last_height       = height;
    ozone->last_scale_factor = gfx_display_get_dpi_scale(p_disp,
-         settings, width, height, false, false);
+         settings, width, height, false, false) * 1.2f;
    ozone->last_thumbnail_scale_factor = settings->floats.ozone_thumbnail_scale_factor;
 
    file_list_initialize(&ozone->selection_buf_old);
@@ -7570,7 +7570,7 @@ static void ozone_set_layout(
    if (!ozone)
       return;
 
-   scale_factor = ozone->last_scale_factor;
+   scale_factor = ozone->last_scale_factor * 1.5;
 
    /* Calculate dimensions */
    ozone->dimensions.header_height                 = HEADER_HEIGHT * scale_factor;
@@ -8273,7 +8273,7 @@ static void ozone_render(void *data,
    /* Check whether screen dimensions or menu scale
     * factor have changed */
    scale_factor = gfx_display_get_dpi_scale(p_disp, settings,
-         width, height, false, false);
+         width, height, false, false) * 1.2f;
    thumbnail_scale_factor = settings->floats.ozone_thumbnail_scale_factor;
 
    if ((scale_factor != ozone->last_scale_factor) ||
@@ -8825,10 +8825,10 @@ static void ozone_draw_header(
          gfx_display_draw_text(
                ozone->fonts.time.font,
                msg,
-               video_width - 85 * scale_factor,
+               video_width - 100 * scale_factor,
                  ozone->dimensions.header_height / 2 
                + ozone->fonts.time.line_centre_offset,
-               video_width,
+               video_width + 15,
                video_height,
                ozone->theme->text_rgba,
                TEXT_ALIGN_RIGHT,
@@ -8850,12 +8850,12 @@ static void ozone_draw_header(
                      status_icon_size,
                      status_icon_size,
                      ozone->icons_textures[powerstate.charging? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_CHARGING : (powerstate.percent > 80)? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_FULL : (powerstate.percent > 60)? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_80 : (powerstate.percent > 40)? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_60 : (powerstate.percent > 20)? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_40 : OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_20],
-                     video_width - (60 + 56) * scale_factor,
+                     video_width - (105 + 56) * scale_factor,
                      0,
                      video_width,
                      video_height,
                      0.0f,
-                     1.0f,
+                     1.5f,
                      col,
                      mymat);
             if (dispctx->blend_end)
@@ -8882,10 +8882,10 @@ static void ozone_draw_header(
       gfx_display_draw_text(
             ozone->fonts.time.font,
             timedate,
-            video_width - (85 * scale_factor) - timedate_offset,
+            video_width - (100 * scale_factor) - timedate_offset,
               ozone->dimensions.header_height / 2 
             + ozone->fonts.time.line_centre_offset,
-            video_width,
+            video_width + 30,
             video_height,
             ozone->theme->text_rgba,
             TEXT_ALIGN_RIGHT,
@@ -8907,12 +8907,12 @@ static void ozone_draw_header(
                   status_icon_size,
                   status_icon_size,
                   ozone->icons_textures[OZONE_ENTRIES_ICONS_TEXTURE_CLOCK],
-                  video_width - (60 + 56) * scale_factor - timedate_offset,
+                  video_width - (140 + 56) * scale_factor - timedate_offset,
                   0,
-                  video_width,
-                  video_height,
+                  video_width + 20,
+                  video_height - 8,
                   0.0f,
-                  1.0f,
+                  1.2f,
                   col,
                   mymat);
          if (dispctx->blend_end)
