diff --git a/src/js2xbox/events.cpp b/src/js2xbox/events.cpp
index 6de93bd..7a28809 100755
--- a/src/js2xbox/events.cpp
+++ b/src/js2xbox/events.cpp
@@ -98,6 +98,7 @@ joypad events::find_event_js(const js_desc** js, js_desc const **out) {
 									id.product == 0x1101 &&
 									id.version == 0x100) 
 									|| strcmp(name, "GO-Super Gamepad") == 0 
+									||strcmp(name, "GO-Advance Gamepad") == 0 // OGA/RK2020
 									||strcmp(name, "GO-Advance Gamepad (rev 1.1)") == 0 ; // RGB10
 
 			bool isOSHController = (strcmp(name, "OpenSimHardware OSH PB Controller") == 0 &&
@@ -202,4 +203,4 @@ void events::print_info_js(const std::string& fname) {
 			}
 		}
 	}
-}
\ No newline at end of file
+}
