diff --git a/menu/menu_setting.c b/menu/menu_setting.c
index 3389534192..071b72ec54 100644
--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -9877,7 +9877,7 @@ static bool setting_append_list(
                parent_group);
          MENU_SETTINGS_LIST_CURRENT_ADD_CMD(list, list_info, CMD_EVENT_MENU_SAVE_CONFIG);
          SETTINGS_DATA_LIST_CURRENT_ADD_FLAGS(list, list_info, SD_FLAG_LAKKA_ADVANCED);
-
+         /*
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_RESET_TO_DEFAULT_CONFIG,
@@ -9887,6 +9887,7 @@ static bool setting_append_list(
                parent_group);
          MENU_SETTINGS_LIST_CURRENT_ADD_CMD(list, list_info, CMD_EVENT_MENU_RESET_TO_DEFAULT_CONFIG);
          SETTINGS_DATA_LIST_CURRENT_ADD_FLAGS(list, list_info, SD_FLAG_LAKKA_ADVANCED);
+         */
 
          CONFIG_ACTION(
                list, list_info,
@@ -9998,7 +9999,7 @@ static bool setting_append_list(
               parent_group);
 #endif
 
-#if defined(HAVE_LAKKA)
+#if defined(HAVE_LAKKA) || defined(HAVE_ODROIDGO2)
 #ifdef HAVE_LAKKA_SWITCH
         CONFIG_ACTION(
                list, list_info,
