diff --git a/menu/menu_setting.c b/menu/menu_setting.c
index 0b9cac7fef..48c23a4346 100644
--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -9770,7 +9770,7 @@ static bool setting_append_list(
                parent_group);
          MENU_SETTINGS_LIST_CURRENT_ADD_CMD(list, list_info, CMD_EVENT_MENU_SAVE_CONFIG);
          SETTINGS_DATA_LIST_CURRENT_ADD_FLAGS(list, list_info, SD_FLAG_LAKKA_ADVANCED);
-
+         /*
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_RESET_TO_DEFAULT_CONFIG,
@@ -9780,6 +9780,7 @@ static bool setting_append_list(
                parent_group);
          MENU_SETTINGS_LIST_CURRENT_ADD_CMD(list, list_info, CMD_EVENT_MENU_RESET_TO_DEFAULT_CONFIG);
          SETTINGS_DATA_LIST_CURRENT_ADD_FLAGS(list, list_info, SD_FLAG_LAKKA_ADVANCED);
+         */
 
          CONFIG_ACTION(
                list, list_info,
