diff --git a/menu/menu_displaylist.c b/menu/menu_displaylist.c
index 8f006cad5b..7e18f49e89 100644
--- a/menu/menu_displaylist.c
+++ b/menu/menu_displaylist.c
@@ -10507,12 +10507,12 @@ unsigned menu_displaylist_build_list(
                   MENU_ENUM_LABEL_SAVE_NEW_CONFIG,
                   MENU_SETTING_ACTION, 0, 0, NULL))
             count++;
-         if (menu_entries_append(list,
+         /*if (menu_entries_append(list,
                   msg_hash_to_str(MENU_ENUM_LABEL_VALUE_RESET_TO_DEFAULT_CONFIG),
                   msg_hash_to_str(MENU_ENUM_LABEL_RESET_TO_DEFAULT_CONFIG),
                   MENU_ENUM_LABEL_RESET_TO_DEFAULT_CONFIG,
                   MENU_SETTING_ACTION, 0, 0, NULL))
-            count++;
+            count++;*/
          break;
       case DISPLAYLIST_PRIVACY_SETTINGS_LIST:
          {
diff --git a/menu/menu_setting.c b/menu/menu_setting.c
index 33bd9a7733..8aef0ee94c 100644
--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -10082,7 +10082,7 @@ static bool setting_append_list(
                parent_group);
          MENU_SETTINGS_LIST_CURRENT_ADD_CMD(list, list_info, CMD_EVENT_MENU_SAVE_CONFIG);
          SETTINGS_DATA_LIST_CURRENT_ADD_FLAGS(list, list_info, SD_FLAG_LAKKA_ADVANCED);
-
+         /*
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_RESET_TO_DEFAULT_CONFIG,
@@ -10092,6 +10092,7 @@ static bool setting_append_list(
                parent_group);
          MENU_SETTINGS_LIST_CURRENT_ADD_CMD(list, list_info, CMD_EVENT_MENU_RESET_TO_DEFAULT_CONFIG);
          SETTINGS_DATA_LIST_CURRENT_ADD_FLAGS(list, list_info, SD_FLAG_LAKKA_ADVANCED);
+         */
 
          CONFIG_ACTION(
                list, list_info,
@@ -10203,7 +10204,7 @@ static bool setting_append_list(
               parent_group);
 #endif
 
-#if defined(HAVE_LAKKA)
+//#if defined(HAVE_LAKKA)
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_REBOOT,
@@ -10222,7 +10223,7 @@ static bool setting_append_list(
                &subgroup_info,
                parent_group);
          MENU_SETTINGS_LIST_CURRENT_ADD_CMD(list, list_info, CMD_EVENT_SHUTDOWN);
-#endif
+//#endif
 
          CONFIG_ACTION(
                list, list_info,
