diff --git a/intl/msg_hash_us.h b/intl/msg_hash_us.h
index 2421a18c88..fd8528a165 100644
--- a/intl/msg_hash_us.h
+++ b/intl/msg_hash_us.h
@@ -10,7 +10,7 @@
 
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_MAIN_MENU,
-   "Main Menu"
+   "ArkOS"
    )
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SETTINGS_TAB,
@@ -4501,7 +4501,7 @@ MSG_HASH(
    )
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_MENU_ADD_CONTENT_ENTRY_DISPLAY_MAIN_TAB,
-   "Main Menu"
+   "ArkOS"
    )
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_MENU_ADD_CONTENT_ENTRY_DISPLAY_PLAYLISTS_TAB,
diff --git a/menu/drivers/ozone.c b/menu/drivers/ozone.c
index 01f9404769..2fb12af6b9 100644
--- a/menu/drivers/ozone.c
+++ b/menu/drivers/ozone.c
@@ -7370,7 +7370,7 @@ static void ozone_set_layout(
    if (!ozone)
       return;
 
-   scale_factor = ozone->last_scale_factor;
+   scale_factor = ozone->last_scale_factor * 1.5;
 
    /* Calculate dimensions */
    ozone->dimensions.header_height                 = HEADER_HEIGHT * scale_factor;
@@ -8607,10 +8607,10 @@ static void ozone_draw_header(
          gfx_display_draw_text(
                ozone->fonts.time.font,
                msg,
-               video_width - 85 * scale_factor,
+               video_width - 100 * scale_factor,
                  ozone->dimensions.header_height / 2 
                + ozone->fonts.time.line_centre_offset,
-               video_width,
+               video_width + 15,
                video_height,
                ozone->theme->text_rgba,
                TEXT_ALIGN_RIGHT,
@@ -8632,12 +8632,12 @@ static void ozone_draw_header(
                      status_icon_size,
                      status_icon_size,
                      ozone->icons_textures[powerstate.charging? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_CHARGING : (powerstate.percent > 80)? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_FULL : (powerstate.percent > 60)? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_80 : (powerstate.percent > 40)? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_60 : (powerstate.percent > 20)? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_40 : OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_20],
-                     video_width - (60 + 56) * scale_factor,
+                     video_width - (105 + 56) * scale_factor,
                      0,
                      video_width,
                      video_height,
                      0.0f,
-                     1.0f,
+                     1.5f,
                      col,
                      mymat);
             if (dispctx->blend_end)
@@ -8664,10 +8664,10 @@ static void ozone_draw_header(
       gfx_display_draw_text(
             ozone->fonts.time.font,
             timedate,
-            video_width - (85 * scale_factor) - timedate_offset,
+            video_width - (100 * scale_factor) - timedate_offset,
               ozone->dimensions.header_height / 2 
             + ozone->fonts.time.line_centre_offset,
-            video_width,
+            video_width + 30,
             video_height,
             ozone->theme->text_rgba,
             TEXT_ALIGN_RIGHT,
@@ -8689,12 +8689,12 @@ static void ozone_draw_header(
                   status_icon_size,
                   status_icon_size,
                   ozone->icons_textures[OZONE_ENTRIES_ICONS_TEXTURE_CLOCK],
-                  video_width - (60 + 56) * scale_factor - timedate_offset,
+                  video_width - (140 + 56) * scale_factor - timedate_offset,
                   0,
-                  video_width,
-                  video_height,
+                  video_width + 20,
+                  video_height - 8,
                   0.0f,
-                  1.0f,
+                  1.2f,
                   col,
                   mymat);
          if (dispctx->blend_end)
diff --git a/menu/menu_setting.c b/menu/menu_setting.c
index 232223281b..5cb190b5bf 100644
--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -9150,6 +9150,7 @@ static bool setting_append_list(
                parent_group);
          SETTINGS_DATA_LIST_CURRENT_ADD_FLAGS(list, list_info, SD_FLAG_LAKKA_ADVANCED);
 
+         /*
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_RESET_TO_DEFAULT_CONFIG,
@@ -9159,6 +9160,7 @@ static bool setting_append_list(
                parent_group);
          MENU_SETTINGS_LIST_CURRENT_ADD_CMD(list, list_info, CMD_EVENT_MENU_RESET_TO_DEFAULT_CONFIG);
          SETTINGS_DATA_LIST_CURRENT_ADD_FLAGS(list, list_info, SD_FLAG_LAKKA_ADVANCED);
+         */
 
          CONFIG_ACTION(
                list, list_info,
@@ -9260,7 +9262,7 @@ static bool setting_append_list(
               parent_group);
 #endif
 
-#if defined(HAVE_LAKKA)
+#if defined(HAVE_LAKKA) || defined(HAVE_ODROIDGO2)
 #ifdef HAVE_LAKKA_SWITCH
         CONFIG_ACTION(
                list, list_info,
diff --git a/retroarch.c b/retroarch.c
index a94c6a7f5e..b3974914f9 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -2446,7 +2446,7 @@ bool command_event(enum event_command cmd, void *data)
          system("(sleep 1 && shutdown -P now) & disown");
 #else
          command_event(CMD_EVENT_QUIT, NULL);
-         system("shutdown -P now");
+         system("sudo shutdown -P now");
 #endif /* HAVE_LAKKA */
 #endif
          break;
@@ -2461,7 +2461,7 @@ bool command_event(enum event_command cmd, void *data)
          system("(sleep 1 && shutdown -r now) & disown");
 #else
          command_event(CMD_EVENT_QUIT, NULL);
-         system("shutdown -r now");
+         system("sudo shutdown -r now");
 #endif /* HAVE_LAKKA */
 #endif
          break;
