diff --git a/intl/msg_hash_us.h b/intl/msg_hash_us.h
index 211e2f9ba3..457473a38e 100644
--- a/intl/msg_hash_us.h
+++ b/intl/msg_hash_us.h
@@ -10,7 +10,7 @@
 
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_MAIN_MENU,
-   "Main Menu"
+   "ArkOS"
    )
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SETTINGS_TAB,
@@ -4443,7 +4443,7 @@ MSG_HASH(
    )
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_MENU_ADD_CONTENT_ENTRY_DISPLAY_MAIN_TAB,
-   "Main Menu"
+   "ArkOS"
    )
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_MENU_ADD_CONTENT_ENTRY_DISPLAY_PLAYLISTS_TAB,
diff --git a/menu/drivers/ozone.c b/menu/drivers/ozone.c
index 03912409a9..8c177859af 100644
--- a/menu/drivers/ozone.c
+++ b/menu/drivers/ozone.c
@@ -7225,7 +7225,7 @@ static void ozone_set_layout(
    if (!ozone)
       return;
 
-   scale_factor = ozone->last_scale_factor;
+   scale_factor = ozone->last_scale_factor * 1.5;
 
    /* Calculate dimensions */
    ozone->dimensions.header_height                 = HEADER_HEIGHT * scale_factor;
@@ -8451,10 +8451,10 @@ static void ozone_draw_header(
          gfx_display_draw_text(
                ozone->fonts.time.font,
                msg,
-               video_width - 85 * scale_factor,
+               video_width - 100 * scale_factor,
                  ozone->dimensions.header_height / 2 
                + ozone->fonts.time.line_centre_offset,
-               video_width,
+               video_width + 15,
                video_height,
                ozone->theme->text_rgba,
                TEXT_ALIGN_RIGHT,
@@ -8476,12 +8476,12 @@ static void ozone_draw_header(
                      status_icon_size,
                      status_icon_size,
                      ozone->icons_textures[powerstate.charging? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_CHARGING : (powerstate.percent > 80)? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_FULL : (powerstate.percent > 60)? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_80 : (powerstate.percent > 40)? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_60 : (powerstate.percent > 20)? OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_40 : OZONE_ENTRIES_ICONS_TEXTURE_BATTERY_20],
-                     video_width - (60 + 56) * scale_factor,
+                     video_width - (105 + 56) * scale_factor,
                      0,
                      video_width,
                      video_height,
                      0.0f,
-                     1.0f,
+                     1.5f,
                      col,
                      mymat);
             if (dispctx->blend_end)
@@ -8508,10 +8508,10 @@ static void ozone_draw_header(
       gfx_display_draw_text(
             ozone->fonts.time.font,
             timedate,
-            video_width - (85 * scale_factor) - timedate_offset,
+            video_width - (100 * scale_factor) - timedate_offset,
               ozone->dimensions.header_height / 2 
             + ozone->fonts.time.line_centre_offset,
-            video_width,
+            video_width + 30,
             video_height,
             ozone->theme->text_rgba,
             TEXT_ALIGN_RIGHT,
@@ -8533,12 +8533,12 @@ static void ozone_draw_header(
                   status_icon_size,
                   status_icon_size,
                   ozone->icons_textures[OZONE_ENTRIES_ICONS_TEXTURE_CLOCK],
-                  video_width - (60 + 56) * scale_factor - timedate_offset,
+                  video_width - (140 + 56) * scale_factor - timedate_offset,
                   0,
-                  video_width,
-                  video_height,
+                  video_width + 20,
+                  video_height - 8,
                   0.0f,
-                  1.0f,
+                  1.2f,
                   col,
                   mymat);
          if (dispctx->blend_end)
diff --git a/menu/menu_setting.c b/menu/menu_setting.c
index 893625038e..fd1495d123 100644
--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -9223,7 +9223,7 @@ static bool setting_append_list(
               parent_group);
 #endif
 
-#if defined(HAVE_LAKKA)
+#if defined(HAVE_LAKKA) || defined(HAVE_ODROIDGO2)
 #ifdef HAVE_LAKKA_SWITCH
         CONFIG_ACTION(
                list, list_info,
diff --git a/retroarch.c b/retroarch.c
index b3040f3299..f17a6885d6 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -12884,7 +12884,7 @@ bool command_event(enum event_command cmd, void *data)
          runloop_msg_queue_push(msg_hash_to_str(MSG_VALUE_SHUTTING_DOWN), 1, 180, true, NULL, MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO);
          command_event(CMD_EVENT_MENU_SAVE_CURRENT_CONFIG, NULL);
          command_event(CMD_EVENT_QUIT, NULL);
-         system("shutdown -P now");
+         system("sudo shutdown -P now");
 #endif
          break;
       case CMD_EVENT_REBOOT:
@@ -12892,7 +12892,7 @@ bool command_event(enum event_command cmd, void *data)
          runloop_msg_queue_push(msg_hash_to_str(MSG_VALUE_REBOOTING), 1, 180, true, NULL, MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO);
          command_event(CMD_EVENT_MENU_SAVE_CURRENT_CONFIG, NULL);
          command_event(CMD_EVENT_QUIT, NULL);
-         system("shutdown -r now");
+         system("sudo shutdown -r now");
 #endif
          break;
       case CMD_EVENT_RESUME:
