diff --git a/retroarch.c b/retroarch.c
index a7e5f5299f..5fddb95dcc 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -4306,7 +4306,7 @@ bool command_event(enum event_command cmd, void *data)
             system("nohup /usr/bin/lakka-shutdown.sh 2>&1 >/dev/null & exit");
 #else
             command_event(CMD_EVENT_QUIT, NULL);
-            system("shutdown -P now");
+            system("sudo shutdown -P now");
 #endif /* HAVE_LAKKA */
 #endif
          }
@@ -4323,7 +4323,7 @@ bool command_event(enum event_command cmd, void *data)
             system("nohup /usr/bin/lakka-reboot.sh 2>&1 >/dev/null & exit");
 #else
             command_event(CMD_EVENT_QUIT, NULL);
-            system("shutdown -r now");
+            system("sudo shutdown -r now");
 #endif /* HAVE_LAKKA */
 #endif
          }
