diff --git a/Makefile b/Makefile
index 5a20bbf..c797a78 100644
--- a/Makefile
+++ b/Makefile
@@ -199,8 +199,19 @@ else ifeq ($(platform), ngc)
    AR = $(DEVKITPPC)/bin/powerpc-eabi-ar$(EXE_EXT)
    ENDIANNESS_DEFINES += -DGEKKO -DHW_DOL -mrvl -mcpu=750 -meabi -mhard-float -DMSB_FIRST
 
-   EXTRA_INCLUDES := -I$(DEVKITPRO)/libogc/include
-   STATIC_LINKING = 1
+# Go-Advance 
+else ifeq ($(platform), goadvance)
+   TARGET := $(TARGET_NAME)_libretro.so
+   fpic := -fPIC
+   SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+   CC = gcc
+   LDFLAGS += $(PTHREAD_FLAGS)
+   FLAGS += $(PTHREAD_FLAGS)
+   FLAGS += -fomit-frame-pointer
+   IS_X86 = 0
+   FLAGS += -DARM -march=armv8-a+crc+simd -mtune=cortex-a35
+   ASFLAGS += -mtune=cortex-a35
+   HAVE_NEON = 1
 else ifeq ($(platform), wii)
    TARGET := $(TARGET_NAME)_libretro_$(platform).a
    CC = $(DEVKITPPC)/bin/powerpc-eabi-gcc$(EXE_EXT)
