diff --git a/yabause/src/nx/main.cpp b/yabause/src/nx/main.cpp
index c19474f0..62efa651 100644
--- a/yabause/src/nx/main.cpp
+++ b/yabause/src/nx/main.cpp
@@ -188,7 +188,7 @@ OSD_struct *OSDCoreList[] = {
 static SDL_Window* wnd;
 static SDL_GLContext glc;
 int g_EnagleFPS = 1;
-int g_resolution_mode = 0;
+int g_resolution_mode = 3;
 int g_keep_aspect_rate = 0;
 int g_scsp_sync = 1;
 int g_frame_skip = 0;
diff --git a/yabause/src/sdl/main.c b/yabause/src/sdl/main.c
index b9214bf6..52f539d6 100644
--- a/yabause/src/sdl/main.c
+++ b/yabause/src/sdl/main.c
@@ -141,7 +141,7 @@ NULL
 #endif
 
 static int fullscreen = 0;
-static int lowres_mode = 0;
+static int lowres_mode = 1;
 
 static char biospath[256] = "\0";
 static char cdpath[256] = "\0";
